### PR TITLE
Add sccache to docker images

### PR DESCRIPTION
**Pull request by driazati**
This adds sccache to Docker images so it can be used later on by specifying the relevant environment variables (`CC` and `CXX` will be picked up by cmake, among others for configuring sccache itself). It is also immediately useful since https://github.com/apache/tvm/blob/main/tests/scripts/ci.py can benefit from cache speedups for rebuilds.

cc @areusch @leandron

### CODE REVIEW

**Review by Mousius — CHANGES_REQUESTED** [review]
Hi @driazati, hugely supportive of this! However, some of the containers don't have Rust installed so it fails with (this was from ci_arm):
```
/install/ubuntu_install_sccache.sh: line 23: cargo: command not found
The command '/bin/sh -c bash /install/ubuntu_install_sccache.sh' returned a non-zero code: 127
```

Definitely would be nice to have a standard base image with Rust / sccache etc to build from. But for this, could you ensure all the containers have Rust added?

**Comment by driazati** [comment]
> Hi @driazati, hugely supportive of this! However, some of the containers don't have Rust installed so it fails with (this was from ci_arm):
> 
> ```
> /install/ubuntu_install_sccache.sh: line 23: cargo: command not found
> The command '/bin/sh -c bash /install/ubuntu_install_sccache.sh' returned a non-zero code: 127
> ```
> 
> Definitely would be nice to have a standard base image with Rust / sccache etc to build from. But for this, could you ensure all the containers have Rust added?

Should be fixed now (broke out the wasmtime dependency from the install rust script since it doesn't work on ARM and it's not needed anyways)

File: docker/Dockerfile.ci_arm [hunk 1]
```diff
@@ -48,3 +48,14 @@ RUN bash /install/ubuntu_install_redis.sh
 # Arm(R) Compute Library
 COPY install/ubuntu_download_arm_compute_lib_binaries.sh /install/ubuntu_download_arm_compute_lib_binaries.sh
 RUN bash /install/ubuntu_download_arm_compute_lib_binaries.sh
+
+# Rust env
```
Comment (Mousius) [inline docker/Dockerfile.ci_arm:52]: Can we put these early as it's a pretty big stage in the Docker build?

File: docker/install/ubuntu_install_wasmtime.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+set -e
+set -u
+set -o pipefail
+set -x
+
+
+export RUSTUP_HOME=/opt/rust
+export CARGO_HOME=/opt/rust
+# this rustc is one supported by the installed version of rust-sgx-sdk
+curl -s -S -L https://sh.rustup.rs -sSf | sh -s -- -y --no-modify-path --profile minimal --default-toolchain stable
```
Comment (Mousius) [inline docker/install/ubuntu_install_wasmtime.sh:28]: I think you copied the wrong bit in here as this installs Rust itself, not wasmtime

**Review by Mousius — CHANGES_REQUESTED** [review]

File: docker/install/ubuntu_install_wasmtime.sh (hunk 2 above)
Comment (driazati) [reply docker/install/ubuntu_install_wasmtime.sh]: ah thanks for catching, it's all fixed now

**Comment by jroesch** [comment]
@driazati can you coordinate with @Mousius or @areusch to update the images after this lands? I think the comments have been addressed from my reading of the PR, going to go ahead and merge.

**Review by jroesch — APPROVED** [review]

**Review by Mousius — APPROVED** [review]
Sorry about the delay, I was going to quickly validate all these build locally before merging but we can do it as part of the docker rebuild :smile_cat:

**Pull request merged**